### ISSUE TITLE
Scaladoc: Link to scoped visibility is always local

### ISSUE THREAD

**Issue by ennru** [scala/bug#11318]
The link to the package for scoped visibility modifiers is always rendered local to the project.

This happens for eg. `clone` in the `scala.Option` class as
https://www.scala-lang.org/api/2.12.7/scala/Option.html contains 

```
<dl class="attributes block"> <dt>Attributes</dt><dd>protected[<a href="../java/lang/index.html" class="extype" name="java.lang">java.lang</a>] 
```
which links to the `java.lang` package locally instead of pointing to external Javadoc.

![bild](https://user-images.githubusercontent.com/458526/50078736-09cd4a00-01e8-11e9-9203-0a4655c65dff.png)

This happens for all Scaladoc, even if linking to external docs works for other parts as seen on https://developer.lightbend.com/docs/api/alpakka/1.0-M1/akka/stream/alpakka/ftp/FtpSettings.html

**Comment by ennru** [comment]
As far as I understand this link is created in `Entity.visibility`

https://github.com/scala/scala/blob/2.13.x/src/scaladoc/scala/tools/nsc/doc/html/page/Entity.scala#L774-L780

and there it shouldn't default to `LinkToTpl`, but decide if `LinkToExternalTpl` should be used.

I don't know what would be the best way to do that. Any guidance?

**Comment by exoego** [comment]
I think it is required to save filters to permalink.
I am considering add query parameter to permalink like 
`https://www.scala-lang.org/api/2.12.7/scala/Option.html?inherited=Option,AnyRef&visibility=public,protected#clone():Object` after #8183 is merged.

**Comment by lrytz** [comment]
@adriaanm blocker for 2.12.9?

**Comment by adriaanm** [comment]
Yes, important for seo of our docs

**Comment by lrytz** [comment]
Looking at the scaladoc codebase for the first time, I'm a still a little lost.

For the issue, I think the assumption was that for qualified private or protected members (`private[X]`), the `X` would always be an internal link, because if you write it in source code, `X` can only be an enclosing entity. However, for inherited members, which are also documented, that is not the case.

```
scala> :pa -raw
// Entering paste mode (ctrl-D to finish)

package p
class K { protected[p] def f = 0 }

// Exiting paste mode, now interpreting.


scala> class C extends p.K
defined class C

scala> :power

scala> typeOf[C].members
res10: $r.intp.global.Scope =
Scope{
  ...
  protected[package p] def f: Int;
  ...
  protected[package lang] def finalize(): Unit;
```

**Comment by eed3si9n** [comment]
Fixed in #8254 by @lrytz
